tests-integration/src/tests/instance_kafka_wal_test.rs

```diff
@@ -74,24 +82,196 @@ async fn test_create_database_and_insert_query(instance: Option<Box<dyn Rebuilda
+        let table = &tables[rng.gen_range(0..tables.len())];
+        for _ in 0..10 {
+            let ts = table.logical_timer.fetch_add(1000, Ordering::Relaxed);
+            let row = make_row(ts, &mut rng);
+            assert_matches!(
+                do_insert(instance, &table.name, row).await,
+                Output::AffectedRows(1)
+            );
+            {
+                // Inserting into the `inserted` vector and inserting into the database are not atomic
+                // which requires us to do a sorting upon checking data integrity.
+                let mut inserted = table.inserted.lock().await;
+                inserted.push(ts);
+            }
+        }
+    }))
+    .await;
+}
+
+async fn ensure_data_exists(tables: &[Arc<Table>], instance: &Arc<Instance>) {
+    futures::future::join_all(tables.iter().map(|table| async {
+        let output = do_query(instance, &table.name).await;
+        let Output::Stream(stream) = output else {
+            unreachable!()
+        };
+        let record_batches = common_recordbatch::util::collect(stream).await.unwrap();
+        let queried = record_batches
+            .into_iter()
+            .flat_map(|rb| {
+                rb.rows()
+                    .map(|row| row[0].as_timestamp().unwrap().value() as u64)
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>();
+        let inserted = table
+            .inserted
+            .lock()
+            .await
+            .iter()
+            .sorted()
+            .cloned()
+            .collect::<Vec<_>>();
+        assert_eq!(queried, inserted);
+    }))
+    .await;
+}
+
+async fn do_create(instance: &Arc<Instance>, table_name: &str) -> Output {
+    execute_sql(
+        instance,
+        &format!(
+            r#"create table greptime.test.{} (
+                    host STRING,
+                    cpu DOUBLE,
+                    memory DOUBLE,
+                    ts timestamp,
+                    TIME INDEX(ts)
+            )"#,
+            table_name
+        ),
+    )
+    .await
+}
+
+async fn do_alter(instance: &Arc<Instance>, table_name: &str, new_table_name: &str) -> Output {
+    execute_sql(
+        instance,
+        &format!("alter table {} rename {}", table_name, new_table_name),
+    )
+    .await
+}
+
+async fn do_insert(instance: &Arc<Instance>, table_name: &str, row: String) -> Output {
+    execute_sql(
+        instance,
+        &format!("insert into test.{table_name}(host, cpu, memory, ts) values {row}"),
+    )
+    .await
+}
+
+async fn do_query(instance: &Arc<Instance>, table_name: &str) -> Output {
+    execute_sql(
+        instance,
+        &format!("select ts from test.{table_name} order by ts"),
+    )
+    .await
+}
+
+async fn execute_sql(instance: &Arc<Instance>, sql: &str) -> Output {
+    instance
+        .do_query(sql, QueryContext::with(DEFAULT_CATALOG_NAME, "test"))
         .await
+        .remove(0)
         .unwrap()
 }
```

Comment:
Consider using the `execute_sql_with` and pass an explicit query context(instead of hard coding)